packages/ifpack/src/Ifpack_Hypre.h

```diff
@@ -140,36 +140,47 @@ class FunctionParameter{
       double_star_func_(funct_name),
       double_star_param_(param1) {}
 
+    //! Integer pointer to list of integer pointers
+    FunctionParameter(Hypre_Chooser chooser, int (*funct_name)(HYPRE_Solver, int**), int ** param1):
+      chooser_(chooser),
+      option_(6),
+	  int_star_star_func_(funct_name),
+	  int_star_star_param_(param1) {}
+
     //! Only method of this class. Calls the function pointer with the passed in HYPRE_Solver
     int CallFunction(HYPRE_Solver solver, HYPRE_Solver precond){
       if(chooser_ == Solver){
-        if(option_ == 0){
-          return int_func_(solver, int_param1_);
-        } else if(option_ == 1){
-          return double_func_(solver, double_param1_);
-        } else if(option_ == 2){
-          return double_int_func_(solver, double_param1_, int_param1_);
-        } else if (option_ == 3){
-          return int_int_func_(solver, int_param1_, int_param2_);
-        } else if (option_ == 4){
-          return int_star_func_(solver, int_star_param_);
-        } else {
-          return double_star_func_(solver, double_star_param_);
-        }
-      } else {
-        if(option_ == 0){
-          return int_func_(precond, int_param1_);
-        } else if(option_ == 1){
-          return double_func_(precond, double_param1_);
-        } else if(option_ == 2){
-          return double_int_func_(precond, double_param1_, int_param1_);
-        } else if(option_ == 3) {
-          return int_int_func_(precond, int_param1_, int_param2_);
-        } else if(option_ == 4) {
-          return int_star_func_(precond, int_star_param_);
+          if(option_ == 0){
+            return int_func_(solver, int_param1_);
+          } else if(option_ == 1){
+            return double_func_(solver, double_param1_);
+          } else if(option_ == 2){
+            return double_int_func_(solver, double_param1_, int_param1_);
+          } else if (option_ == 3){
+            return int_int_func_(solver, int_param1_, int_param2_);
+          } else if (option_ == 4){
+            return int_star_func_(solver, int_star_param_);
+          } else if (option_ == 5){
+            return double_star_func_(solver, double_star_param_);
+          } else {
```

Comment:
It would be nicer to test `option_ == 6`, and to have an `else` that catches unsupported `option_` values.  However, I would accept this PR without that.

Reply:
I thought I needed to rebase on devlop branch and just did that. But I see that this PR was already approved before I force updated by local branch